src/components/settings/PresetButtons.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+  import {
+    presetStores,
+    presets,
+    activePreset
+  } from '../../js/store';
+  import { importStores } from '../../js/storage';
+  import Button from 'smelte/src/components/Button';
+  import PresetButton from './PresetButton.svelte';
+
+  let presetAmount = $presets.length;
+
+  let isDeleting = false;
+  let activeNumber = $activePreset;
+  $: activePreset.set(activeNumber);
+
+  async function updatePreset(presetNumber: number) {
+    if (presetNumber !== activeNumber) {
+      importStores(JSON.stringify($presets[presetNumber - 1]));
+      activeNumber = presetNumber;
+      return;
+    }
+
+    const presetsData = $presets;
+    presetsData[presetNumber - 1] = Object.fromEntries(presetStores.map(store => [store.name, store.get()]));
+    presets.set(presetsData);
+  }
+
+  async function deletePreset(presetNumber: number) {
+    const presetsData = $presets;
+    presetsData.splice(presetNumber - 1, 1);
+    presets.set(presetsData);
+    presetAmount -= 1;
+  }
+
+  async function addPreset() {
+    const presetsData = $presets;
+    presetsData.push(Object.fromEntries(presetStores.map(store => [store.name, store.defaultValue])));
+    presets.set(presetsData);
+    presetAmount += 1;
+  }
+
+</script>
+
+
+<div class="grid grid-cols-2 gap-2 py-1">
+  {#each Array(presetAmount) as _, i}
+    <PresetButton 
+      activeNumber = { activeNumber }
```

Comment:
two things here
1. you can do `{ activeNumber }`
2. you probably should use event dispatchers instead (basically event listeners) and listen to `on:eventName`. Check out https://svelte.dev/tutorial/component-events

Reply:
Ok should be fixed now.